src/tcp.rs

```diff
@@ -421,4 +427,36 @@ mod tests {
         assert_eq!(deserialized, header);
         assert_eq!(re_deserialized, header);
     }
+    #[test]
+    fn try_clone(){
+        let mut close = Arc::new(Mutex::new(()));
+        thread::scope(|s| {
+            let mut close_clone = close.clone();
+            let jh = s.spawn(move ||{
+                let listener = TcpListener::bind("127.0.0.1:34254").unwrap();
+                listener.accept().and_then(|(tcp,addr)| {
+                    while close_clone.try_lock().is_ok(){}
+                    Ok(())
+                }).unwrap();
+            });
+            let new_stream = TcpStream::connect("127.0.0.1:34254").unwrap();
+            let mut transport = Transport{ tid: 1, uid: 2, stream: new_stream };
```

Comment:
`transport` does not need to be `mut`

Reply:
There is a stream write which is expected to fail here: https://github.com/AShriki/modbus-rs/blob/master/src/tcp.rs#L451
But requires `transport `to be `mut `to keep rustc happy.